RecoMuon/TrackerSeedGenerator/plugins/MuonHLTSeedMVAClassifier.cc

```diff
@@ -0,0 +1,285 @@
+  const int nSeedsMaxB_;
+  const int nSeedsMaxE_;
+
+  const bool rejectAll_;
+  const bool isFromL1_;
+  const int minL1Qual_;
+  const double baseScore_;
+
+  double getSeedMva(const pairSeedMvaEstimator& pairMvaEstimator,
+                    const TrajectorySeed& seed,
+                    GlobalVector global_p,
+                    edm::Handle<l1t::MuonBxCollection>& h_L1Muon,
+                    int minL1Qual_,
+                    edm::Handle<reco::RecoChargedCandidateCollection>& h_L2Muon,
+                    bool isFromL1_,
+                    double baseScore_);
+};
+
+MuonHLTSeedMVAClassifier::MuonHLTSeedMVAClassifier(const edm::ParameterSet& iConfig)
+    : t_Seed_(consumes<TrajectorySeedCollection>(iConfig.getParameter<edm::InputTag>("src"))),
+      t_L1Muon_(consumes<l1t::MuonBxCollection>(iConfig.getParameter<edm::InputTag>("L1Muon"))),
+      t_L2Muon_(consumes<reco::RecoChargedCandidateCollection>(iConfig.getParameter<edm::InputTag>("L2Muon"))),
+
+      etaEdge_(iConfig.getParameter<double>("etaEdge")),
+      mvaCutB_(iConfig.getParameter<double>("mvaCutB")),
+      mvaCutE_(iConfig.getParameter<double>("mvaCutE")),
+
+      doSort_(iConfig.getParameter<bool>("doSort")),
+      nSeedsMaxB_(iConfig.getParameter<int>("nSeedsMaxB")),
+      nSeedsMaxE_(iConfig.getParameter<int>("nSeedsMaxE")),
+
+      rejectAll_(iConfig.getParameter<bool>("rejectAll")),
+      isFromL1_(iConfig.getParameter<bool>("isFromL1")),
+      minL1Qual_(iConfig.getParameter<int>("minL1Qual")),
+      baseScore_(iConfig.getParameter<double>("baseScore")) {
+  if (!rejectAll_) {
+    mvaFileB_ = iConfig.getUntrackedParameter<edm::FileInPath>("mvaFileB");
+    mvaFileE_ = iConfig.getUntrackedParameter<edm::FileInPath>("mvaFileE");
+
+    mvaScaleMeanB_ = iConfig.getUntrackedParameter<std::vector<double>>("mvaScaleMeanB");
+    mvaScaleStdB_ = iConfig.getUntrackedParameter<std::vector<double>>("mvaScaleStdB");
+    mvaScaleMeanE_ = iConfig.getUntrackedParameter<std::vector<double>>("mvaScaleMeanE");
+    mvaScaleStdE_ = iConfig.getUntrackedParameter<std::vector<double>>("mvaScaleStdE");
+
+    mvaEstimator_ = std::make_pair(std::make_unique<SeedMvaEstimator>(mvaFileB_, mvaScaleMeanB_, mvaScaleStdB_),
+                                   std::make_unique<SeedMvaEstimator>(mvaFileE_, mvaScaleMeanE_, mvaScaleStdE_));
+  }
+
+  produces<TrajectorySeedCollection>();
+}
+
+// -- method called on each new Event
+void MuonHLTSeedMVAClassifier::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  auto result = std::make_unique<TrajectorySeedCollection>();
+
+  if (rejectAll_) {
+    iEvent.put(std::move(result));
+    return;
+  }
+
+  if (doSort_ && nSeedsMaxB_ <= 0 && nSeedsMaxE_ <= 0) {
+    iEvent.put(std::move(result));
+    return;
+  }
+
+  if (!doSort_ && mvaCutB_ > 1. && mvaCutE_ > 1.) {
+    iEvent.put(std::move(result));
+    return;
+  }
+
+  edm::ESHandle<TrackerGeometry> trkGeom;
+  iSetup.get<TrackerDigiGeometryRecord>().get(trkGeom);
```

Comment:
Direct call of function EventSetupRecord::get(ESHandle&) is deprecated and should be replaced with a call to EventSetup::getHandle(ESGetToken&). To use ESGetToken see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#In_ED_module To get data with the token see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_from_EventSetup_wit